# Knowledge Base
All lab work-related knowledge we gather (not code)

## Collections

- [Awesome Single-cell](https://github.com/seandavi/awesome-single-cell): list of almost all single-cell tools
- [Single-cell RNA-seq analysis best practices](https://github.com/theislab/sc-best-practices-ce) from Theis lab
- [Single-Cell Tutorial](https://github.com/theislab/single-cell-tutorial) from Theis lab
- [Single-Cell Introduction](https://hemberg-lab.github.io/scRNA.seq.course/) course from Hemberg lab

## Tools to use

### Validated

### Didn't pass validation

- [CellAssign](https://github.com/irrationone/cellassign): automated annotation based on provided markers. Requires to install tensorflow on R. Didn't manage to install it on the server (@vpetukhov).
- [Garnett](https://cole-trapnell-lab.github.io/garnett/): automated annotation based on provided markers. Tried on epilepsy data, quality was too low to use (@vpetukhov). Maybe could work on simpler datasets.

### Not tested

- [scfind](https://scfind.sanger.ac.uk/): tool from Hemberg lab, which allow to look for markers in the published atlases. Should simplify annotation.
- [SingleR](https://bioconductor.org/packages/devel/bioc/html/SingleR.html): Performs unbiased cell type recognition from single-cell RNA sequencing data, by leveraging reference transcriptomic datasets of pure cell types to infer the cell of origin of each single cell independently.
